src/lsm/lsm_cursor.c

```diff
@@ -889,6 +889,80 @@ err:	__clsm_leave(clsm);
 }
 
 /*
+ * __clsm_next_random --
+ *	WT_CURSOR->next method for the LSM cursor type when configured with
+ * next_random.
+ */
+static int
+__clsm_next_random(WT_CURSOR *cursor)
+{
+	WT_CURSOR_LSM *clsm;
+	WT_CURSOR *c;
+	WT_DECL_RET;
+	WT_LSM_CHUNK *chunk;
+	WT_SESSION_IMPL *session;
+	uint64_t checked_docs, i, rand_chunk, rand_doc, total_docs;
+	const char *cfg[3];
+
+	clsm = (WT_CURSOR_LSM *)cursor;
+
+	CURSOR_API_CALL(cursor, session, next, NULL);
+	WT_CURSOR_NOVALUE(cursor);
+	WT_ERR(__clsm_enter(clsm, false, false));
+
+	cfg[0] = WT_CONFIG_BASE(session, WT_SESSION_drop);
+	cfg[1] = "next_random=true";
+	cfg[2] = NULL;
+
+retry:	total_docs = 0;
+	/* We need to know roughly how many docs are in the tree */
+	for (i = clsm->lsm_tree->nchunks; i > 0; i--) {
+		total_docs += clsm->lsm_tree->chunk[i-1]->count;
+	}
```

Comment:
Would this be more readable:
{code}
for (i = 0; i < clsm->lsm_tree->nchunks; i++) {
       total_docs += clsm->lsm_tree->chunk[i]->count;
}
{code}